services/analyzer/sequelize-tables-analyzer.js

```diff
@@ -79,52 +79,138 @@ function hasIdColumn(fields, primaryKeys) {
     || _.includes(primaryKeys, 'id');
 }
 
+// NOTICE: Check the foreign key's reference unicity
+function checkUnicity(primaryKeys, uniqueIndexes, columnName) {
+  const isUnique = uniqueIndexes !== null
+    && uniqueIndexes.find((indexColumnName) =>
+      indexColumnName.length === 1 && indexColumnName.includes(columnName));
+
+  const isPrimary = _.isEqual([columnName], primaryKeys);
+  return { isPrimary, isUnique };
+}
```

Comment:
IsUnique does not look to take into consideration `partial unique indexes`.
For conditional unique indexes we should set `hasMany` not `hasOne`:
<img width="495" alt="Screenshot 2019-12-17 at 09 02 36" src="https://user-images.githubusercontent.com/664261/70976518-6047bf80-20ac-11ea-9936-7a37e86a9bdb.png">
![Uploading Screenshot 2019-12-17 at 09.02.49.png…]()

Reply:
When I test it with planActivations, I can create as many planActivations as long as I have only one with finishedAt not set. This seems to be the behavior we are looking for, furthermore, I couldn't find any query allowing us to list the partial indexes.

Reply:
Maybe it is difficult to fix the issue, but the expected relationship here is `Billings.hasMany(models.planActivations, ...`.
See our internal server code ;).